gather-scheduler.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const PAUSE_AFTER_LOAD = 3000;
+
+class GatherScheduler {
+
+  static _pauseAfterLoad() {
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, PAUSE_AFTER_LOAD);
+    });
+  }
+
+  static _runPhase(gatherers, gatherFun) {
+    return gatherers.reduce((chain, gatherer) =>
```

Comment:
so I think we are in the clear from infinite handing since we are tapping into the .then() =>

now the confusing was in the crazy whitespace with single line => functions, I would prefer for readability to use explicit `()` and `{}` around here :) ... since the end of the arrow function and start of the second parm is kinda hard to see and depends on whitespace :palm_tree: